src/Interpreters/Cache/SLRUFileCachePriority.cpp

```diff
@@ -269,17 +269,22 @@ bool SLRUFileCachePriority::collectCandidatesForEviction(
 
     stat += probationary_stat;
 
-    LOG_TEST(log, "Collected {} to evict from probationary queue. Total size: {}",
-             res.size(), probationary_stat.total_stat.releasable_size);
+    LOG_TEST(log, "Collected {} to evict from probationary queue "
+             "with total size: {} (result: {}). "
+             "Desired size: {}, desired elements count: {}, current state: {}",
+             probationary_stat.total_stat.releasable_count,
+             probationary_stat.total_stat.releasable_size, res.size(),
+             desired_probationary_size, desired_probationary_elements_num,
+             probationary_queue.getStateInfoForLog(lock));
 
     chassert(!max_candidates_to_evict || res.size() <= max_candidates_to_evict);
     chassert(res.size() == stat.total_stat.releasable_count);
 
     if (max_candidates_to_evict && res.size() >= max_candidates_to_evict)
         return probationary_limit_satisfied;
 
-    const auto desired_protected_size = getRatio(max_size, size_ratio);
-    const auto desired_protected_elements_num = getRatio(max_elements, size_ratio);
+    const auto desired_protected_size = getRatio(desired_size, size_ratio);
+    const auto desired_protected_elements_num = getRatio(desired_elements_count, size_ratio);
```

Comment:
After adding more logs in this PR and seeing the error still present in stress test I found the actual bug is here. Will add a unit test for this. Will mark this PR as non-draft after all stress tests pass.